For review with Android

KirillSmirnow opened the pull request:
In addition to all the code as in #89, this PR contains the latest android module code, though the checks fail

KirillSmirnow committed:
Initialize Kotlin Multiplatform Application

KirillSmirnow committed:
Refactor a little

KirillSmirnow committed:
Set up Spring Boot

KirillSmirnow committed:
Configure CI/CD

KirillSmirnow committed:
Fix Android shared module build

KirillSmirnow committed:
Fix server deployment

KirillSmirnow committed:
Add Spring modules and shared module

On branch MR-1-user

KirillSmirnow committed:
Add Spring modules and shared module

On branch MR-1-user

KirillSmirnow committed:
Add User @Entity

On branch MR-1-user

KirillSmirnow committed:
Add User @Entity

On branch MR-1-user

KirillSmirnow committed:
Add Identifier @Entity

On branch MR-1-user

KirillSmirnow committed:
Add Identifier @Entity

On branch MR-1-user

KirillSmirnow committed:
Add tests for user sign-up

1. User creation test
2. Identifier creation test

On branch MR-1-user

KirillSmirnow committed:
Add tests for user sign-up

1. User creation test
2. Identifier creation test

On branch MR-1-user

KirillSmirnow committed:
Add 'SMS is sent on sign-up' test

On branch MR-1-user

KirillSmirnow committed:
Add 'SMS is sent on sign-up' test

On branch MR-1-user

KirillSmirnow committed:
Fix shared lib dependency

On branch MR-1-user

KirillSmirnow committed:
Fix shared lib dependency

On branch MR-1-user

KirillSmirnow committed:
WIP: implement user sign-up

On branch MR-1-user

KirillSmirnow committed:
WIP: implement user sign-up

On branch MR-1-user

KirillSmirnow committed:
Implement user sign-up

6-digit verification code generation by means of Apache Commons Lang

On branch MR-1-user

KirillSmirnow committed:
Implement user sign-up

6-digit verification code generation by means of Apache Commons Lang

On branch MR-1-user

KirillSmirnow committed:
Add tests for sign-up with existing phone number

On branch MR-1-user

KirillSmirnow committed:
Add tests for sign-up with existing phone number

On branch MR-1-user

KirillSmirnow committed:
Declare signIn method and add some tests for it

On branch MR-1-user

KirillSmirnow committed:
Declare signIn method and add some tests for it

On branch MR-1-user

KirillSmirnow committed:
Implement user sign-in

On branch MR-1-user

KirillSmirnow committed:
Implement user sign-in

On branch MR-1-user

KirillSmirnow committed:
Add some badges to readme

On branch MR-1-user

KirillSmirnow committed:
Add some badges to readme

On branch MR-1-user

KirillSmirnow committed:
Create Session Entity

On branch MR-4-session

KirillSmirnow committed:
Add tests for session start

On branch MR-4-session

KirillSmirnow committed:
Include kotlinx-datetime library

The library provides multiplatform support of date/time classes

On branch MR-4-session

KirillSmirnow committed:
Implement session starting

On branch MR-4-session

KirillSmirnow committed:
Add test that user is enabled after phone has been verified

On branch MR-4-session

KirillSmirnow committed:
Add test of non-existent phone number

On branch MR-4-session

KirillSmirnow committed:
Add test that locked user won't be able to sign in

On branch MR-1-user

KirillSmirnow committed:
Define verifySession method and cover it with tests

On branch MR-4-session

KirillSmirnow committed:
Implement session verification

On branch MR-4-session

KirillSmirnow committed:
Complete expired session test

Use Mockito (v3.6) to mock static method of Clock class
so as to have the ability to change system time

On branch MR-4-session

KirillSmirnow committed:
Merge branch 'main' of github.com:ItmoDreamTeam/MyRest into MR-4-session

KirillSmirnow committed:
Fix shared lib structure

Directory with source code in shared module was named like 'org.itmo.../'
instead of 'org/itmo.../' which was unusual

On branch project-structure-fix

KirillSmirnow committed:
Merge branch 'main' into MR-4-session

KirillSmirnow committed:
Add Restaurant Entity and Repository

On branch MR-5-restaurant

KirillSmirnow committed:
Add Restaurant Entity and Repository

On branch MR-5-restaurant

KirillSmirnow committed:
Fix packages/directories

On branch MR-4-session

KirillSmirnow committed:
Add RestaurantTable entity and repository

On branch MR-8-tables

Rznov committed:
Define registerRestaurant method and cover it with tests

Rznov committed:
Rename RestaurantInfo, add more tests for registerRestaurant

Rznov committed:
Implement restaurant registration

Rznov committed:
Declare updateRestaurant method, add tests for it

Rznov committed:
Implement updateRestaurant method, add more tests to it

Rznov committed:
Declare getById method, add tests for it

Rznov committed:
Implement getById method

KirillSmirnow committed:
Add employees entities

Rznov committed:
Declare search method, add tests for it

KirillSmirnow committed:
Make User comparable

KirillSmirnow committed:
Add waiters atrribute to RestaurantTable

Rznov committed:
Implement search method, refactor tests for it

KirillSmirnow committed:
Add Reservation entity

KirillSmirnow committed:
Merge branch 'main' into MR-4-session

Rznov committed:
Refactor RequestServiceImpl and method tests

Rznov committed:
Resolving conflicts

KirillSmirnow committed:
Declare NotificationService interface

and add stub implementation

KirillSmirnow committed:
Declare NotificationService interface

and add stub implementation

KirillSmirnow committed:
Declare NotificationService interface

and add stub implementation

KirillSmirnow committed:
Merge branch 'main' into MR-4-session

KirillSmirnow committed:
Configure uploading of test results and coverage report

KirillSmirnow committed:
Merge branch 'main' into MR-4-session

KirillSmirnow committed:
Declare inviteEmployee method and cover it with tests

On branch MR-11-employees

KirillSmirnow committed:
Declare inviteEmployee method and cover it with tests

On branch MR-11-employees

KirillSmirnow committed:
Implement employee invitation and add some tests for it

On branch MR-11-employees

KirillSmirnow committed:
Implement employee invitation and add some tests for it

On branch MR-11-employees

KirillSmirnow committed:
Configure employee notification when invitation created

On branch MR-11-employees

KirillSmirnow committed:
Configure employee notification when invitation created

On branch MR-11-employees

KirillSmirnow committed:
Split employee status into two

Now employee is described by means of two statuses. One from the perspective
of the user, another from the perspective of the restaurant.

A single status wouldn't provide the required flexibily.

On branch MR-11-employees

KirillSmirnow committed:
Split employee status into two

Now employee is described by means of two statuses. One from the perspective
of the user, another from the perspective of the restaurant.

A single status wouldn't provide the required flexibily.

On branch MR-11-employees

KirillSmirnow committed:
Design the rest of the EmployeeService methods

On branch MR-11-employees

KirillSmirnow committed:
Design the rest of the EmployeeService methods

On branch MR-11-employees

Rznov committed:
Fix methods implementation/tests

KirillSmirnow committed:
Implement employees retrieval

Implement restaurants retrieval by a user (employee) and employees (users)
retrieval by a restaurant. Cover the methods with tests.

On branch MR-11-employees

KirillSmirnow committed:
Implement employees retrieval

Implement restaurants retrieval by a user (employee) and employees (users)
retrieval by a restaurant. Cover the methods with tests.

On branch MR-11-employees

KirillSmirnow committed:
Cover employee status updating with tests

On branch MR-11-employees

KirillSmirnow committed:
Cover employee status updating with tests

On branch MR-11-employees

KirillSmirnow committed:
Implement employee status updating

Implement employee status updating by user him/herself and by
restaurant managers.

On branch MR-11-employees

KirillSmirnow committed:
Implement employee status updating

Implement employee status updating by user him/herself and by
restaurant managers.

On branch MR-11-employees

Rznov committed:
Fix RestaurantService/tests for it

Rznov committed:
Add toRestaurantInfo method

KirillSmirnow committed:
Add User to Profile transformation method

On branch MR-4-session

KirillSmirnow committed:
Add attributes to Profile DTO

And move User Role to shared module

On branch MR-4-session

Rznov committed:
Refactor RestaurantUpdateInfo, RestaurantServiceImpl

KirillSmirnow committed:
Refactor Session class a little

On branch MR-4-session

KirillSmirnow committed:
Merge branch 'main' into MR-5-restaurant

KirillSmirnow committed:
Reformat code

On branch MR-5-restaurant

KirillSmirnow committed:
Merge branch 'main' into MR-11-employees

KirillSmirnow committed:
Resolve TODOs

Insert Profile and RestaurantInfo objects into EmployeeInfo
Fix creation of EmployeeInfo and tests

On branch MR-11-employees

KirillSmirnow committed:
Resolve TODOs

Insert Profile and RestaurantInfo objects into EmployeeInfo
Fix creation of EmployeeInfo and tests

On branch MR-11-employees

KirillSmirnow committed:
Declare TableService

On branch MR-8-tables

KirillSmirnow committed:
Add tests for TableService

On branch MR-8-tables

KirillSmirnow committed:
Implement TableService

On branch MR-8-tables

KirillSmirnow committed:
Fix tables creation test

Object types to compare were different

On branch MR-8-tables

KirillSmirnow committed:
Fix RestaurantTable validation rules and add messages

On branch MR-8-tables

KirillSmirnow committed:
Refactor EmployeeServiceImpl

On branch MR-11-employees

KirillSmirnow committed:
Refactor EmployeeServiceImpl

On branch MR-11-employees

KirillSmirnow committed:
Add UserServiceMock and RestaurantServiceMock

Mock objects provide testing of EmployeeService in isolation
from other services, without depending on their actual implementations.

On branch MR-11-employees

KirillSmirnow committed:
Add UserServiceMock and RestaurantServiceMock

Mock objects provide testing of EmployeeService in isolation
from other services, without depending on their actual implementations.

On branch MR-11-employees

KirillSmirnow committed:
Rename StubSmsService using the term Fake

Fake objects are those having working implementations,
but not suitable for production

KirillSmirnow committed:
Merge branch 'main' into MR-11-employees

KirillSmirnow committed:
Rename StubNotificationService using the term Fake

On branch MR-11-employees

KirillSmirnow committed:
Rename StubNotificationService using the term Fake

On branch MR-11-employees

KirillSmirnow committed:
Merge branch 'main' of github.com:ItmoDreamTeam/MyRest into MR-8-tables

KirillSmirnow committed:
Fix table updating tests

On branch MR-8-tables

KirillSmirnow committed:
Merge branch 'MR-11-employees' of github.com:ItmoDreamTeam/MyRest into MR-8-tables

KirillSmirnow committed:
Define TableWaiterService

Also:
 * Add RestaurantServiceMock to use for restaurant table tests
 * Prepare TableWaiterServiceTest class

On branch MR-8-tables

KirillSmirnow committed:
Cover TableWaiterService with tests

On branch MR-8-tables

KirillSmirnow committed:
Implement TableWaiterService

Also add required mock objects for testing

On branch MR-8-tables

Rznov committed:
Update register method declaration, added tests

KirillSmirnow committed:
Merge branch 'main' of github.com:ItmoDreamTeam/MyRest into MR-8-tables

Rznov committed:
Update register method implementation

Rznov committed:
Declare updateStatus method, add tests for it

Rznov committed:
Implement updateStatus method, add NotificationServiceMock

KirillSmirnow committed:
Add returning objects from remove methods

Methods removeTable and removeWaiter will now return table and waiter
objects that are removed

On branch MR-8-tables

Rznov committed:
Refactor register method ad tests

Rznov committed:
Refactor register method and test

Rznov committed:
Fix register method and tests

Rznov committed:
Fix update method

KirillSmirnow committed:
Move same restaurant check to service layer

On branch MR-8-tables

KirillSmirnow committed:
Replace table's name with table number

Table number is of type Int as it is common to distinguish tables
in a restaurant by their order numbers.

On branch MR-8-tables

KirillSmirnow committed:
Add tests for unique table numbers

On branch MR-8-tables

Rznov committed:
Add gitignore for ios client

Rznov committed:
Remove scene delegate

Rznov committed:
Set up the project, remove unnecessary files

KirillSmirnow committed:
Define shared RestaurantController

KirillSmirnow committed:
Enable AndroidX support

KirillSmirnow committed:
Add RestaurantSearchActivity

Rznov committed:
Implement RestaurantListViewController, add RestaurantCell, LoadingFotter

Rznov committed:
Add pod for DI container, add composition root

Rznov committed:
Add custom VerticalCollectionView

Rznov committed:
Implement SearchTextField

KirillSmirnow committed:
Fix compilation errors, refactor and reformat code

On branch MR-14-Restaurant_verification

KirillSmirnow committed:
Configure Java version (11) for android build

KirillSmirnow committed:
Create ViewModel and bind it to Activity

On branch MR-22-api-consumption

KirillSmirnow committed:
Configure http client

Android
 * Enable Internet permission
 * Set HttpClient default properties

Shared
 * Add ClientProperties object

On branch MR-22-api-consumption

KirillSmirnow committed:
Add server web security configuration

On branch MR-22-api-consumption

KirillSmirnow committed:
Set android app name and icon

On branch MR-22-api-consumption

Rznov committed:
Implement SearchView, Implement RestaurantListViewController presentation logic

KirillSmirnow committed:
Implement search API method

On branch MR-22-api-consumption

KirillSmirnow committed:
Configure response deserialization on client side

On branch MR-22-api-consumption

Rznov committed:
Define RestaurantListRouter

KirillSmirnow committed:
Move Android ktor client dependency from android to shared/android module

On branch MR-22-api-consumption

KirillSmirnow committed:
Move HttpClient init into shared module

On branch MR-22-api-consumption

KirillSmirnow committed:
Add Ktor client for iOS

On branch MR-22-api-consumption

KirillSmirnow committed:
Add coroutines dependency for iOS

On branch MR-22-api-consumption

KirillSmirnow committed:
Configure shared module build for iOS simulator (iosX64)

On branch MR-22-api-consumption

KirillSmirnow committed:
Fix coroutines dependencies

Add serialization (json) dependency

On branch MR-22-api-consumption

Rznov committed:
Declare SignIn scene, implement PhoneTextField

Rznov committed:
Implement RestaurantListRouter

KirillSmirnow committed:
Add UserController

Include dependency for Jackson to deserialize Kotlin data classes

On branch MR-31-user-controller

KirillSmirnow committed:
Add UserController

Include dependency for Jackson to deserialize Kotlin data classes

On branch MR-31-user-controller

KirillSmirnow committed:
Configure user authentication by access token

Implement current user retrieval

On branch MR-31-user-controller

KirillSmirnow committed:
Configure user authentication by access token

Implement current user retrieval

On branch MR-31-user-controller

KirillSmirnow committed:
Add getMe method to UserController

Add plugin that implicitly makes Kotlin classes and methods,
marked with Spring annotations, open.

On branch MR-31-user-controller

KirillSmirnow committed:
Add getMe method to UserController

Add plugin that implicitly makes Kotlin classes and methods,
marked with Spring annotations, open.

On branch MR-31-user-controller

KirillSmirnow committed:
Wrap session verification into InvalidBearerTokenException

On branch MR-31-user-controller

KirillSmirnow committed:
Wrap session verification into InvalidBearerTokenException

On branch MR-31-user-controller

KirillSmirnow committed:
Add UserClient

On branch MR-31-user-controller

KirillSmirnow committed:
Add UserClient

On branch MR-31-user-controller

KirillSmirnow committed:
Fix request body serialization: set content-type header

On branch MR-31-user-controller

KirillSmirnow committed:
Fix request body serialization: set content-type header

On branch MR-31-user-controller

KirillSmirnow committed:
Implement providing of access token in client requests

On branch MR-31-user-controller

KirillSmirnow committed:
Implement providing of access token in client requests

On branch MR-31-user-controller

KirillSmirnow committed:
Fix controllers/clients naming

Remove exception handling on client level

On branch MR-31-user-controller

KirillSmirnow committed:
Fix controllers/clients naming

Remove exception handling on client level

On branch MR-31-user-controller

Rznov committed:
Implement PhoneTextField

KirillSmirnow committed:
Create registration method of RestaurantController

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Make UserService Transactional

This fixes implicit saving data to DB

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Change restaurant registration HTTP method to PUT

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Add API method for updating restaurant status

This method is accessible by user with ADMIN role only

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Add API method for searching among all restaurants by admin

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Add API method for updating restaurant info

Configure permission evaluation (Task MR-34)

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Implement RestaurantPermissionEvaluator

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Add new RestaurantController methods to RestaurantClient

On branch MR-33-restaurant-controller

KirillSmirnow committed:
Fix restaurant creator status bug

Refactor restaurant registration method

On branch MR-35-restaurant-creator-status-bug

KirillSmirnow committed:
Fix path of session start endpoint

On branch MR-31-user-controller

KirillSmirnow committed:
Merge branch 'main' into MR-33-restaurant-controller

Rznov committed:
Implement SignInViewController

KirillSmirnow committed:
Merge branch 'main' of github.com:ItmoDreamTeam/MyRest into MR-33-restaurant-controller

Rznov committed:
Implement RestaurantListPresenter, RestaurantListInteractor

Rznov committed:
Implement RestaurantListView method

Rznov committed:
Resolve dependencies

Rznov committed:
Fix phoneTextField, attach SignInScreen screenshot

KirillSmirnow committed:
Merge branch 'main' into MR-39-Restaurant-search

Commit:
Add RestaurantClient invocation

The invocation currently results in serialization error:

Uncaught Kotlin exception: kotlinx.serialization.SerializationException: Serializer for class 'RestaurantInfo' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
On Kotlin/Native explicitly declared serializer should be used for interfaces and enums without @Serializable annotation

On branch MR-42-native-serialization

KirillSmirnow committed:
Configure objects to JSON serialization on native platforms

On branch MR-42-native-serialization

KirillSmirnow committed:
Revert "Add RestaurantClient invocation"

That commit was added for testing purposes

This reverts commit 2ca2535d0df43c7c6f86903c0205ab8154eccdcd.

Rznov committed:
Merge branch 'main' of https://github.com/ItmoDreamTeam/MyRest into MR-39-Restaurant-search

Rznov committed:
Improve RestaurantsListViewController

Rznov committed:
Merge branch 'MR-39-Restaurant-search' of https://github.com/ItmoDreamTeam/MyRest into MR-39-Restaurant-search

Rznov committed:
Improve RestaurantsListViewController

KirillSmirnow committed:
Merge branch 'main' into MR-39-Restaurant-search

AlexIvchenko committed:
MR-21: converters (#52)

* Create ModelView assemble API
Refactor UserService within ModelViewAssembler

* Refactor RestaurantService within ModelViewAssembler

* Refactor TableService within ModelViewAssembler

* Refactor EmployeeService within ModelViewAssembler

* Give conventional name to UserProfileAssembler

* Give method more conventional name

KirillSmirnow committed:
MR-44: DTOs serialization (#45)

* Configure LocalDateTimeSerializer

* Mark all DTOs with @Serializable

* Limit JVM heap size

* Configure Jackson serializers for kotlinx LocalDateTime

On branch MR-44-dtos-serialization

* Make computable attribute of EmployeeInfo transient

On branch MR-44-dtos-serialization

* Use kotlinx serialization for Android client

Remove previously used Gson library.
kotlinx serialization will be used for all clients
so as to share common serializers.

On branch MR-44-dtos-serialization

* Create a single file for kotlinx serializers in config package

In shared module same as in server module

* Rename KotlinxSerializers package

KirillSmirnow committed:
Add 'build on macOS' CI job (#61)

AlexIvchenko committed:
MR-15: Table Reservation (#40)

* Split single time field to time-range to provide more flexible reservation mechanism
Drop related DB constraint

* Implement ReservationService:
- submit/reject reservation by user
- approve/start/complete reservation by responsible manager

* Implement ReservationViewService to be used in controller level
Add DTO for Reservation

* Fix tests

* Remove redundant @Param annotations

* Split large ugly string into small tasteful pieces

* Clarify query method name
Remove annoying comment

* Fix typo

* Sanitize test

* Configure report generation pipeline

* FreeMarker processes HTML/CSS template
* itextpdf converts HTML/CSS to PDF
* Add API method

On branch MR-26-reports

* Implement reservations report generation

On branch MR-26-reports

* Sort reservation entries

Add some styles to report

On branch MR-26-reports

* Add API methods to ReservationController

On branch MR-47

* Add getRestaurantsOfUser API method

On branch MR-47

* Fix ReservationController paths

On branch MR-47

* Add EmployeeController

On branch MR-47

* Configure authorization of employee update methods

On branch MR-47

* Implement reservations fetch API methods

On branch MR-47

* Configure authorization of ReservationController methods

On branch MR-47

* Add RestaurantTableController

On branch MR-47

* Fix compilation

* Remove security checks on domain service layer

* Implement reservation view assembler
Add user, table, manager dtos to ReservationInfo

* Refactor test

* Move reservation to separate package
Reduce tests spring context coupling

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

Rznov committed:
MR-54: SignUp page (#59)

* Fix app dependencies

* Add pods

* Implement SignUpViewController

* Refactor RestaurantListRouter

* Refactor SignUpViewController

* Remove unused pod

* Implement SignUpInteractor, declare SignUpPresenter

* Implement SignUpPresenter

* Add CodeTextField to SignUpViewController

* Refactor PhoneTextField

* Rename PhoneTextField

* Refactor SignUpView

* Resolve dependencies for SignUpScene

* Fix dependencies

* Fix typos

* Implement Verification code sending

* Refactor SignUpViewController

* Refactor MaskTextField

* Replace text property to getText()

* Refactor typo

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

KirillSmirnow committed:
MR-23: errors handling (#56)

* Implement exception handling structure

* Implement simple UserException conversion

* Implement interpolatable UserException conversion

* Add interpolation with multiple parameters test

* Move test config to package level

* Implement ConstraintViolationException conversion

* Implement AccessDeniedException conversion

* Add error messages for Restaurant entity

* Add error messages for RestaurantTable entity

* Fix test table numbers

Regarding that numbers must be <= 100

* Move UserException messages to errors.yml

* Implement ktor client response validation and error handling

* Improve AccessTokenProvider

Even if access token is null, send it to server to receive and handle 401 Unauthorized

* Rename Error to ServerError because of conflict in Swift

* Make global objects, required for components registration, thread local

* Declare ClientException as throwable from clients' methods

* Upgrade Kotlin version

Which allows to remove @ExperimentalStdlibApi

* Cache errors of AccessDeniedExceptionToErrorsConverter

* Implement base ErrorHandler<CONTEXT>

* Move reservation errors messages to errors.yml

KirillSmirnow committed:
MR-63: notifications (#65)

* Implement FirebaseNotificationService

* Update notifications content

* Configure Firebase app init

* Configure FCM for Android app

* Implement reservation time interval formatting

* Fix token unregistration log message

* Implement notifications retrieval and update

* Add NotificationClient

Fix pagination

* Implement messaging token registration

KirillSmirnow committed:
MR-58: SMS (#62)

* Create RealSmsService

* Add SmsMessage entity

* Implement RealSmsService

* Shorten sign-up verification message to save money

* Implement SMS daily limit check

* Fix sign-up verification SMS test

* Configure SMS service credentials

* Implement Russian text messages sending (though, transliterated)

* Add errors messages

* Add incorrect phone error message

Rznov committed:
MR-64: User info page (#67)

* Declare UserInfoViewController

* Implement UserInfoViewController

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

Rznov committed:
MR-57: Errors handling (#69)

* Implement AccessTokenProviderImpl

* Implement AccessTokenProviderImpl

* Fix project file

* Implement IOSErrorHandler, ToSignInRouter

* Register AccessTokenProviderImpl

* Refactor IOSErrorHandler

* Fix IOSErrorHandlerDelegate

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

KirillSmirnow committed:
Configure Google app creds (#71)

AlexIvchenko committed:
Implement simple sign in activity
Add skeleton of sign up activity
Add manual dependency injection
Auto upgrade of gradle dependencies

Rznov committed:
MR-66: Restaurant list page business logic (#72)

* Add comments to RestaurantListViewController

* Add method to IOSErrorHandler

* Add missing dependencies for RestaurantListInteractorImpl

* Fix AppDependencies

* Implement restaurantListDidRequestUserInfo

* Declare onCompleted RestaurantList scene methods

* Refactor RestaurantListRouter

* IOSErrorHandler

* Clean up the code

* Clean up the code

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

KirillSmirnow committed:
MR-55: clients (#73)

* Add employee, table and reservation clients

* Fix DTOs serialization

* Fix DTOs serialization

* Fix employee status updating

* Convert EmployeeInfo active property to method due to deserialization issues

Rznov committed:
MR-74: Authorisation routing (#77)

* Add comments to RestaurantListViewController

* Add method to IOSErrorHandler

* Add missing dependencies for RestaurantListInteractorImpl

* Fix AppDependencies

* Implement restaurantListDidRequestUserInfo

* Declare onCompleted RestaurantList scene methods

* Refactor RestaurantListRouter

* IOSErrorHandler

* Clean up the code

* Clean up the code

* Move MaskTextField

* Declare EnterCodeViewController

* Implement EnterCodeViewController.storyboard

* Implement EnterCodeViewController

* Refactor SignInViewController

* Declare SignInInteractor

* Conform to swiftlint

* Implement SignInInteractor and SignInPresenter

* Implement SignInRouter

* Implement SignInRouter

* Declare VerificationCodeInteractor

* Declare SignInViewDataDelegate

* Implement VerificationCodePresenter

* Add ContextDataDelegate

* Refactor SignUpScene

* Refactor SignUpScene

* Add extension for SignUpViewController: ContextDataDelegate

* Refactor routing from SignInScene

* Add interactor to VerificationCodeViewController

* Implement VerificationCodeRouter

* Clean up the code

* Implement SignUpRouter

* Resolve dependencies

* Refactor ToSignInRouter

* Refсtor SignInInteractor

* Refactor VerificationCodeViewController

* Refactor VerificationCodeInteractor

* Clean up UserInfoViewController

* Declare ProfileDataDelegate

* Extend UserInfoViewController: ProfileDataDelegate

* Fix App dependencies

* Fix NavBar bug

Co-authored-by: Kirill Smirnov <smirnowku@gmail.com>

KirillSmirnow committed:
MR-78: photos (#79)

* Add avatar and photos attributes to Restaurant

* Implement attachment downloading

KirillSmirnow committed:
MR-75: logging (#76)

* Add ServiceLogger

* Mark ThrowableToErrorsConverterImpl as Service to improve logging

Rznov committed:
MR-48: Restaurant info page (#81)

* Implement RestaurantInfoViewController

* Refactor RestaurantViewModel

* Implement RestarauntIdDataDelegate

* Remove RestarauntIdDataDelegate, add RestarauntInfoDataDelegate

* Replace RestaurantViewModel to RestaurantInfo, add VerticalCollectionViewDelegate

* Refactor RestaurantListRouter

* Refactor RestaurantInfoScene

* Refactor RestaurantListScene

* Add registerButton to RestaurantInfoVC

* Refactor PhotosCollectionView

* Add SDWebImage to podfile

* Update podfile

* Add SDWebImage with SPM

* Set restaurant images

* Set restaurant avatar

* Implement restaurant photos

* Implement HalfSizePresentationController

* Declare RestaurantInfoRouter

AlexIvchenko committed:
Add skeleton for navigation
Add DI using hilt
Enable data binding
Fix java versions (hilt doesn't work with 11)
Add some dummy fragments

AlexIvchenko committed:
Implement sign-in on nav approach
Add toolbar
Add material components
Set theme to NoActionBar (to see the back icon?)

AlexIvchenko committed:
Store access-token and last used phone in SharedProperties
Implement auto-recovering session logic
Remove old activities from manifest

AlexIvchenko committed:
Implement restaurant list view
Add profile view
Remove obsolete debug search activities
Auto-move to restaurant search if user is signed in

KirillSmirnow committed:
MR-80: web client (#84)

* Create restaurant search component

* Configure CORS

* Implement error handler and access token provider

* Get restaurant list on page opening

* Work around of ContentPage<E> deserialization in Kotlin for JS

* Fix initial search state

* Search item & style

* Add RootComponent

* Add profile page

* Implement sign in

* Implement redirection to sign in and back

* Implement header component

* Implement SignUpComponent

* Add user restaurants list on profile page

* Improve restaurant search list items

* Implement user's reservations list

Rznov committed:
MR-82: Book page (#83)

* Declare BookViewController

* Implement RestaurantInfoRouter

* Implement BookInteractor and BookPresenter

* Implement pickers

* Add utils for Date

* Add more util for Date

* Implement table reservation

* Implement BookRouter

* Configure registerButton

* Fix shemes

KirillSmirnow committed:
MR-86: website deployment and README (#87)

* Configure website deployment

* Add Production, Build and Deploy sections to README

* Make FCM initialization optional (for local/debug envs)

* Fix Build section of README

* Add 'Development with IntelliJ IDEA' section to README

KirillSmirnow committed:
Remove non-existent activity from AndroidManifest

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/error/ServerError.kt:
```diff
@@ -0,0 +1,18 @@
+    val developerMessage: String? = null,
+) {
+    companion object {
+        const val UNKNOWN_ERROR_KEY = "unknown"
```
Тогда уж тоже по-русски ;)

KirillSmirnow replied:
Не, `unknown` это же key/ключ ошибки, по аналогии с `user.not-found`

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/MessagingClient.kt:
```diff
@@ -0,0 +1,10 @@
+
+interface MessagingClient {
+
+    @Throws(CancellationException::class, ClientException::class)
```
Не уверен, что указание эксепшнов здесь нужно

KirillSmirnow replied:
Нужно для ios, к сожалению

KirillSmirnow replied:
В @Throws мы указываем какие типы исключений являются ожидаемыми, все остальные -- убьют приложение. CancellationException тоже обязательное. Это только для ios.

Walingar replied:
Понял, не знал

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/MessagingClientImpl.kt:
```diff
@@ -0,0 +1,18 @@
+import org.itmodreamteam.myrest.shared.AccessTokenProvider.Companion.provideAccessToken
+import org.itmodreamteam.myrest.shared.HttpClientProvider
+
+class MessagingClientImpl : MessagingClient {
```
unused class

KirillSmirnow replied:
Клиенты будут его использовать попозже

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/NotificationClient.kt:
```diff
@@ -0,0 +1,15 @@
+
+interface NotificationClient {
+
+    @Throws(CancellationException::class, ClientException::class)
```
аналогично

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/NotificationClientImpl.kt:
```diff
@@ -0,0 +1,29 @@
+import org.itmodreamteam.myrest.shared.Pageable
+import org.itmodreamteam.myrest.shared.Pageable.Companion.addPageableParameters
+
+class NotificationClientImpl : NotificationClient {
```
unused class

KirillSmirnow replied:
Клиенты будут его использовать попозже

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/NotificationContent.kt:
```diff
@@ -0,0 +1,20 @@
+    )
+
+    companion object {
+        fun fromMap(data: Map<String, String>): NotificationContent {
```
unused method

KirillSmirnow replied:
Клиенты будут его использовать попозже

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/messaging/NotificationContent.kt:
```diff
@@ -0,0 +1,20 @@
+    val title: String,
+    val text: String,
+) {
+    fun toMap(): Map<String, String> = mapOf(
```
Map<String, String> -> Map<Enum, String>

KirillSmirnow replied:
Объект данного класса преобразуется в Map<String, String>, передаётся через Firebase на клиент, и из Map<String, String> создаётся объект

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/restaurant/EmployeePosition.kt:
```diff
@@ -0,0 +1,6 @@
+package org.itmodreamteam.myrest.shared.restaurant
+
+enum class EmployeePosition(val representation: String) {
+    MANAGER("Менеджер"),
```
А еще правильней было бы сделать Bundle ;) В котором по ключу вы бы получали строку нужную. Вдруг захотите потом на разных языках иметь приложение

KirillSmirnow replied:
Это да)

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/restaurant/ReservationClientImpl.kt:
```diff
@@ -0,0 +1,70 @@
+    private val client = HttpClientProvider.provide()
+
+    override suspend fun getReservationsOfRestaurant(restaurantId: Long, date: LocalDate): List<ReservationInfo> {
+        return client.get {
```
Много копипасты вида client.get { ... provideAccessToken() }

KirillSmirnow replied:
AccessToken не везде нужен, некоторые запросы публичные, и кажется правильнее разделять публичные запросы и приватные

Walingar replied:
Тогда можно как раз вынести экстеншн методы privateRequest, publicRequest?

Walingar replied:
Если они начнут отличаться не только в токене?)

Walingar commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/AccessTokenProvider.kt:
```diff
@@ -0,0 +1,19 @@
+
+    val accessToken: String?
+
+    @ThreadLocal
```
Главное потом не ошибиться в коррутинах и передавать эту проперти в контекст. А то не забывайте, что корутины запускаются в разных потоках

Walingar commented on ios/ios/ApplicationLayer/AppDelegate.swift:
```diff
@@ -0,0 +1,23 @@
+import Dip
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
```
Чур swift я ревьюить не буду)

Walingar commented on server/src/main/kotlin/org/itmodreamteam/myrest/server/controller/PageUtil.kt:
```diff
@@ -0,0 +1,10 @@
+import org.springframework.data.domain.Page
+
+object PageUtil {
+    fun <E> toContentPage(page: Page<E>): ContentPage<E> {
```
Можно сделать val Page<E>.contentPage = ...

Walingar commented on server/src/main/kotlin/org/itmodreamteam/myrest/server/controller/ReservationController.kt:
```diff
@@ -0,0 +1,88 @@
+
+    @PutMapping("/reservations/{reservationId}/approve")
+    @PreAuthorize("hasPermission(#reservationId, 'Reservation', 'write-manager')")
+    fun approve(@PathVariable reservationId: Long): ReservationInfo {
```
`return` -> `=`

Walingar commented on server/src/main/kotlin/org/itmodreamteam/myrest/server/model/restaurant/Manager.kt:
```diff
@@ -0,0 +1,14 @@
+
+    constructor() : super()
+
+    constructor(restaurant: Restaurant, user: User) : super(restaurant, user)
```
class Manager(restaurant: Restaurant, user: User): Employee(restaurant, user)

Walingar replied:
Не получится так с классами рядом сделать? или тут спринг себя плохо ведет?

KirillSmirnow replied:
Ему обязательно нужен пустой конструктор

Walingar commented on server/src/main/kotlin/org/itmodreamteam/myrest/server/security/permission/ReservationPermissionEvaluator.kt:
```diff
@@ -0,0 +1,34 @@
+
+    override val targetType: String = "Reservation"
+
+    override fun hasPermission(authentication: UserAuthentication, targetId: Long, permission: String): Boolean {
```
Лучше, конечно в месте, где спринг вам отдает permission мапить их в Enum и стараться меньше использовать String

Walingar commented on android/src/main/java/org/itmodreamteam/myrest/android/RestaurantSearchViewModel.kt:
```diff
@@ -0,0 +1,32 @@
+    private val restaurantClient: RestaurantClient
+) : ViewModel() {
+    private val _restaurants = MutableLiveData<List<RestaurantInfo>>()
+    val restaurants: LiveData<List<RestaurantInfo>> = _restaurants
```
Класс, что так делаешь!

Walingar commented on android/src/main/java/org/itmodreamteam/myrest/android/RestaurantSearchViewModel.kt:
```diff
@@ -0,0 +1,32 @@
+            val search = restaurantClient.search(keyword, Pageable(0, 10))
+            // TODO debug x4
+            val test = mutableListOf<RestaurantInfo>()
+            for (i in (0..3)) {
```
Скобки лишние

Walingar commented on android/src/main/java/org/itmodreamteam/myrest/android/MyRestApplication.kt:
```diff
@@ -0,0 +1,8 @@
+import dagger.hilt.android.HiltAndroidApp
+
+@HiltAndroidApp
+class MyRestApplication : Application() {
```
{} не нужны

Walingar commented on js/src/main/kotlin/user/ProfileComponent.kt:
```diff
@@ -0,0 +1,123 @@
+    private fun getMyRestaurants() = GlobalScope.launch {
+        try {
+            val restaurants = props.restaurantClient.getRestaurantsOfUser()
+            setState(State(state.me, restaurants))
```
state.copy(restaurants = restaurants)

Walingar replied:
Насколько я знаю, в kotlin-react больше любят делать state интерфейсом и проперти var, чтобы тут вызывать setState { restaurants = restaurants }

KirillSmirnow replied:
Да, так было бы лучше

Walingar commented on js/src/main/kotlin/user/SignInComponent.kt:
```diff
@@ -0,0 +1,132 @@
+class SignInComponent(props: Props) : RComponent<SignInComponent.Props, SignInComponent.State>(props) {
+
+    init {
+        state = State(null)
```
Я думаю, так не очень хорошо делать, лучше использовать state который предоставляет родитель

KirillSmirnow commented on shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/attachment/AttachmentMetadata.kt:
```diff
@@ -0,0 +1,13 @@
+    val name: String,
+    val type: String,
+) {
+    fun url(): String = "http://${Server.host}:${Server.port}/attachments/$objectId/$name"
```
Так можно сделать, если сконфигурировать ktor client deserializer, чтобы он игнорировал поле url. Иначе возникают ошибки, то в android, то в ios...

Walingar replied:
Тогда можно сделать extension val ;)

KirillSmirnow committed:
Add seatingCharts to Restaurant (#93)

KirillSmirnow committed:
Remove SMS daily limit

AlexIvchenko committed:
Store access token
Refresh session
Add stub navigation

AlexIvchenko committed:
Add getById method for RestaurantClient (#94)

* Add getById method for RestaurantClient

* Disable auth for public method

AlexIvchenko committed:
Add restaurant details fragment
Add assisted DI support

AlexIvchenko committed:
Implement reservation of table

AlexIvchenko committed:
Implement view of reservations

AlexIvchenko committed:
Add settings fragment
Fix singletons
Dynamic menu depending on global employee mode

AlexIvchenko committed:
Beautify login page

AlexIvchenko committed:
Implement mocked signUp
Implement logout
Setup navigation root destinations

AlexIvchenko committed:
Add ability to update profile (#95)

* Add ability to update profile

* Fix tests

AlexIvchenko committed:
Fix opening restaurant after it had been reserved

AlexIvchenko committed:
Fix endpoint body (#96)

AlexIvchenko committed:
Bypass signUp in user is not registered
Remove signUp fragment
Add ability to change profile data
Global timer for signIn action

AlexIvchenko committed:
Add ability to get employee by its id (#97)

AlexIvchenko committed:
Implement employee mode

AlexIvchenko committed:
Add ability to filter manager's reservations

AlexIvchenko committed:
Sanitize time intervals

AlexIvchenko committed:
Make urls clickable on details
Remove urls when it's not needed

AlexIvchenko committed:
Beautify reservation card: date, time
Add info about table to reservation card
